### DISCUSSION TITLE
kcna23 TOC<->TAG (follow up) "How can we support each other" - shared vocabulary, model(s),

### DISCUSSION THREAD

**Discussion by halcyondude** [cncf/toc#1210] [Ideas]
# Context and beginnings

_...our tale begins at circular tables that could be (likely are) used for award ceremonies, lunch, weddings, "Under the Sea" dances, and in our case, a lively meeting between TOC members and TAG co-chairs at KubeCon 2023_

---

At the TAG Chairs + TOC Meeting at KubeCon North America 2023 (kcna23), we had a healthy discussion around how we can better support each other and the projects in our ecosystem. 

During that discussion ([video](https://www.youtube.com/watch?v=79fFiOO7XzQ&t=889s)) we talked about building consensus around nomenclature and models for types of projects, users, and contributors so that we can assess

* project community health and sustainability and End User adoption
* technical capability, purpose, domain, areas, scenarios, and {up,down}stream dependencies
* needs as identified by the project and it's maintainers

As there are ongoing efforts in TAG Contributor Strategy on a project maturity model (TODO Link), we should solicit input, and work with TAG Contributor Strategy to understand how these efforts may be complimentary, and how/where any consensus based artifacts may be preserved and used.  

## Behavior based Classification

Captured as a separate Discussion issue ([cncf/toc/discussions/1219](https://github.com/cncf/toc/discussions/1219)) so that this issue can track what we'll do with the output of that discussion.

## Protecting Contributor Privacy

I mentioned the need to protect contributor privacy as we assess our ecosystem(s) and their communities. This has been moved to separate Discussion issue ([cncf/toc/discussions/1218](https://github.com/cncf/toc/discussions/1218)) so that this issue can track what we'll do with the output of that discussion.

## (proposed) Next Steps

**Logistics**

- [ ] ServiceDesk issue for Google Drive for TAGs.
- [ ] move docs --> new location + inform #tag-chairs + email.
- [ ] Get on calendar: TAG Contributor Strategy

**Tasks**

- [x] Solicit input and feedback from TAG Chairs & TOC @ KubeCon meeting
  - [x] Create https://github.com/cncf/toc/discussions/1218 (privacy)
  - [x] Create https://github.com/cncf/toc/discussions/1219 (project and contributor types)
  - [x] Create https://github.com/cncf/toc/discussions/1220 (End User Feedback (TAB) -> TAGs)
- [ ] Solicit input and feedback from domain experts and others 
  - [ ] CHAOSS - https://chaoss.community
  - [ ] TODO - https://todogroup.org)
  - [ ] ???
- [ ] Solicit input and discuss at TAG Contributor Strategy
- [ ] Report back --> toc/tagchairs

**Comment by halcyondude** [comment]
slack convo for posterity: https://cloud-native.slack.com/archives/CSCPTLTPE/p1699573726198079

**Comment by TheFoxAtWork** [comment]
this looks like it wraps a lot of topics together... I want to tease this apart a bit to focus on two core topics so we can make progress against each:
1. Re-evaluation of the "domains", "verticals", and "workloads" construct to better convey capability groups projects _align_ with. Breaking these into sub-categories for more granular reasoning.
2. End User Adoption - related to the End User TAB. to me this means we need to ensure the TAGs have clear bi-directional communication with the End User TAB and its groups, as appropriate.

@halcyondude does this align with what you were capturing or am i thinking of something else?

**Reply by halcyondude** [reply]
Updates incoming, have been resource constrained and travelling.

**Reply by halcyondude** [reply]
@TheFoxAtWork I've cleaved 3 issues off this one (privacy, terms, end user feedback), leaving this to track next steps. https://github.com/cncf/toc/discussions/1220 could use some detail, who might be a good person to drive this?

**Comment by halcyondude** [comment]
@TheFoxAtWork it does align - (1.) is the activity and outcome that can be informed by this discussion topic.

> 1. Re-evaluation of the "domains", "verticals", and "workloads" construct to better convey capability groups projects align with. Breaking these into sub-categories for more granular reasoning.

Using this issue to track this, as the model(s) can inform a shared vocabulary. I look forward to understanding what other models we might use in our discourse and subsequent discussions, as well as discovering if others have found this model to be useful, or if there are other ways to model and frame the same.

> 2. End User Adoption - related to the End User TAB. to me this means we need to ensure the TAGs have clear bi-directional communication with the End User TAB and its groups, as appropriate.

Moved to: [cncf/toc/discussions/1220](https://github.com/cncf/toc/discussions/1220)

**Comment by TheFoxAtWork** [comment]
Agreed, they can inform a shared vocabulary. It could also be leveraged as an index of potential considerations and constraints across each area and assist in scoping project alignment, implementation, and resolution of needs or concerns across all constructs.
So lets start by defining each of these - i'll provide my perception of these in the context of cloud native.

Domain - a defined field of technical focus, engineering, research, or study where the organized entity has influence over.  Domains may be inclusive of sub-domains which further divide the focus of the domain into skill-unique areas, i.e. Security as a domain can consist of network security, identity and access management, policy, detection engineering, chaos engineering, secure development, etc. 

Vertical - a categorization of market, offerings, or business interest that organizations cater to or operate within, i.e. Financial Services, Healthcare, Government, Information Technology, etc.

Workloads - a categorization of compute resources (applications, services, configuration, etc.) to accomplish a task or outcome, i.e. business applications, AI, e-commerce, streaming media, etc.

**Reply by halcyondude** [reply]
The implementation for #1711 will draw from this. Thank you for responding in detail with your thoughts.

**Comment by TheFoxAtWork** [comment]
This may be worth bringing to TAG CS for additions to [Cloud Native Glossary](https://contribute.cncf.io/resources/glossary/)

**Comment by halcyondude** [comment]
I think this is a good candidate for an Initiative. Will link back to here after drafting and submitting as an issue.

**Comment by halcyondude** [comment]
I've proposed a new Initiative (using the post TAG Reboot structuring) that should draw from, and be informed by, this Discussion.

[[Initiative]: CNCF Project Capabilities Badging Framework #1711)](https://github.com/cncf/toc/issues/1711)
